go/tasks/plugins/presto/client/presto_status.go

```diff
@@ -0,0 +1,41 @@
+package client
+
+import (
+	"context"
+	"strings"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/svc"
+	"github.com/lyft/flytestdlib/logger"
+)
+
+// This type is meant only to encapsulate the response coming from Presto as a type, it is
+// not meant to be stored locally.
+const (
+	PrestoStatusUnknown   svc.CommandStatus = "UNKNOWN"
```

Comment:
I tried this but ran into a problem:
```
enumer: can't handle non-integer constant type CommandStatus
go/tasks/plugins/presto/client/presto_status.go:10: running "enumer": exit status 1
```
I think this might be why the `qubole_status.go` doesn't use the generator.

Reply:
Here is an example that works:
https://play.golang.org/p/DOSciZ7joY1

I needed to implement UnmarshalJSON()...

Reply:
@EngHabu Sorry, I'm still a little confused as I think this example you posted  uses:
```
type TransitionType int
```
where the type is `int`. In my cause, the status is a `string` and based on my attempts above, it doesn't seem to work on non-integer types. Am I missing something?

Reply:
Does it need to be a string though?

Reply:
Ok, this is not a big deal... if it doesn't seem obvious to change, that's fine...